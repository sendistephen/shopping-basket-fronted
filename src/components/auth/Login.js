import React from "react";

const Login = () => {
	return (
		<section class="main center">
			<div class="container">
				<div class="main-content-wrapper">
					<div class="main-content-holder">
						<h2>
							Shopping made <span>easy</span>
						</h2>
						<p>
							Manage your shopping with ease. Create, read, update <br /> and delete your shopping
							basket.
						</p>
					</div>
					<div class="card">
						<p>Have an account? Login here.</p>
						<form>
							<input class="form-control" type="email" placeholder="Email" />
							<input class="form-control" type="password" placeholder="Password" />
							<input class="btn btn-register" type="submit" value="Login" />
						</form>
					</div>
				</div>
			</div>
		</section>
	);
};

export default Login;
