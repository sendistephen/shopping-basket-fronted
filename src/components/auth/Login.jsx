import React, { Component } from 'react';

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return <h1>Hello login</h1>;

}

export default Login;
