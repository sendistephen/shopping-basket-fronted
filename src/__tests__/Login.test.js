import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Login from '../components/auth/Login';

describe('Login component', () => {
  const wrapper = shallow(<Login />);
  const preventDefault = jest.fn();

  it('renders properly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders email input', () => {
    expect(shallow(<Login />).find('#email').length).toEqual(1);
  });

  it('renders password input', () => {
    expect(shallow(<Login />).find('#password').length).toEqual(1);
  });

  describe('Email input', () => {
    it('should respond to change event and change the state of the Login Component', () => {
      const wrapper = shallow(<Login />);
      const userEmail = wrapper.find('#email');
      userEmail.simulate('change', {
        target: { name: 'email', value: 'test@example.com' },
      });
      userEmail.value = 'test@example.com';
      expect(userEmail.value).toEqual('test@example.com');
    });
  });

  describe('Password input', () => {
    it('should respond to change event and change the state of the Login Component', () => {
      const wrapper = shallow(<Login />);
      const userPassword = wrapper.find('#password');
      userPassword.simulate('change', {
        target: { name: 'password', value: 'secret' },
      });
      userPassword.value = 'secret';
      expect(userPassword.value).toEqual('secret');
    });
  });
});
