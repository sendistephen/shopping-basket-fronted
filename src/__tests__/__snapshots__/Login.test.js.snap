// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders properly 1`] = `
<div
  className="container"
>
  <div
    className="card"
    style={
      Object {
        "margin": "70px auto",
        "width": "40%",
      }
    }
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title text-center font-weight-bold"
      >
        Shopping Basket
      </h5>
      <p
        className="card-text text-center lead mb-5"
      >
        Welcome back. Please log in here.
      </p>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group form-rounded"
        >
          <input
            className="form-control form-rounded"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control form-rounded"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </div>
        <button
          className="btn btn-register btn-block"
          type="submit"
        >
          Log In
        </button>
        <hr
          className="my-4"
        />
        <p
          className="lead text-center"
          style={
            Object {
              "color": "#ABBBC8",
            }
          }
        >
          Don't have an account?
           
          <Link
            style={
              Object {
                "color": "#0071E2",
                "fontWeight": "400",
              }
            }
            to="/"
          >
            Sign up
          </Link>
        </p>
      </form>
    </div>
  </div>
</div>
`;
